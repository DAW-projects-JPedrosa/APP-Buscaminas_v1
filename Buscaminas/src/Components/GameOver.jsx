import { useState} from 'react';

function GameOver({win}) {
  const [isVisible, setIsVsible] = useState(true);

  function closeMessage(){
    setIsVsible(false);
  }

    return isVisible?(
        <>
          <div id='game-over-msg' className={win? 'win': 'lost'}>
            <div className='game-over'>{win? 'Has ganado': 'Has perdido'}</div>
            <button className='close' onClick={closeMessage}>Cerrar</button>
          </div>
        </>
      ) : null;
}

export default GameOver;