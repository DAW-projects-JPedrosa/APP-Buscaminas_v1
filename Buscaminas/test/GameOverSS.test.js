import React from 'react';
import Renderer from "react-test-renderer";
import GameOver from '../src/Components/GameOver'; 

it("Renders a message of game over when the player wins", () => {
    //Arrange
    const win = true;

    //Act
    const gameOverComponent = Renderer.create(
        <GameOver win={win}></GameOver>
    )

    //Assert
    let tree = gameOverComponent.toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders a message of game over when the player loses", () => {
    //Arrange
    const win = false;

    //Act
    const gameOverComponent = Renderer.create(
        <GameOver win={win}></GameOver>
    )

    //Assert
    let tree = gameOverComponent.toJSON();
    expect(tree).toMatchSnapshot();
})
