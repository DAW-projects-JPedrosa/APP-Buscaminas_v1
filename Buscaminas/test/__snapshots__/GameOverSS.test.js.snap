// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a message of game over when the player loses 1`] = `
<div
  className="lost"
  id="game-over-msg"
>
  <div
    className="game-over"
  >
    Has perdido
  </div>
  <button
    className="close"
    onClick={[Function]}
  >
    Cerrar
  </button>
</div>
`;

exports[`Renders a message of game over when the player wins 1`] = `
<div
  className="win"
  id="game-over-msg"
>
  <div
    className="game-over"
  >
    Has ganado
  </div>
  <button
    className="close"
    onClick={[Function]}
  >
    Cerrar
  </button>
</div>
`;
